File: modules/apps/frontend-taglib/frontend-taglib/src/main/resources/META-INF/resources/diff_version_comparator/components/Filter.js
/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

import ClayForm, {ClayInput} from '@clayui/form';
import ClayIcon from '@clayui/icon';
import React, {useEffect, useRef} from 'react';

export default function Filter({onChange, query}) {
	const inputRef = useRef();

	useEffect(() => {
		inputRef.current.focus();
	}, [query]);

	return (
		<ClayForm.Group>
			<ClayInput.Group>
				<ClayInput.GroupItem prepend>
					<ClayInput
						onChange={onChange}
						placeholder={Liferay.Language.get('search')}
						ref={inputRef}
						type="text"
						value={query}
					/>
				</ClayInput.GroupItem>

				<ClayInput.GroupItem append shrink>
					<ClayInput.GroupText>
						<ClayIcon symbol="search" />
					</ClayInput.GroupText>
				</ClayInput.GroupItem>
			</ClayInput.Group>
		</ClayForm.Group>
	);
}

File: modules/apps/portal-search/portal-search-api/src/main/java/com/liferay/portal/search/filter/Filter.java
/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.portal.search.filter;

import org.osgi.annotation.versioning.ProviderType;

/**
 * @author André de Oliveira
 */
@ProviderType
public interface Filter extends com.liferay.portal.kernel.search.filter.Filter {

	@Override
	public default <T> T accept(
		com.liferay.portal.kernel.search.filter.FilterVisitor<T>
			filterVisitor) {

		return (T)accept((FilterVisitor)filterVisitor);
	}

	public <T> T accept(FilterVisitor<T> filterVisitor);

}